bin/import_xml2obd.py

```diff
@@ -236,16 +346,45 @@ def upload_resources(remote_site, api_key, jsonfile, resource_directory):
             continue
         res = rec['resources'][0]
         source = resource_directory + '/' + res['url'].split('/')[-1]
-        with open(source) as f:
-            try:
-                rc = site.action.resource_patch(
-                    id=res['id'],
-                    upload=(os.path.basename(source), f))
-            except ckanapi.errors.CKANAPIError:
-                traceback.print_exc()
-                print(rec)
-                print("failed upload " + os.path.basename(source))
-                continue
+        fname='/'.join(['resources', res['id'], res['url'].split('/')[-1]])
+
+        obj = dest.get_obj(fname.lower())
+        skip = False
+        srcmd5 = md5str(source)
+        if not srcmd5:
+            raise Exception('not found ' + source)
+        md5 = binascii.unhexlify(srcmd5)
+        if obj:
+            objmd5 = base64md5str(obj.properties.content_settings.content_md5)
+            if objmd5 == srcmd5:
+               skip = True
+            else:
+                print (objmd5, srcmd5)
+        if target_pkg and not skip:
+            for res in target_pkg['resources']:
+                rmd5 = res.get('hash', None)
+                if rmd5[:4] == 'md5-':
+                    rmd5 = base64md5str(rmd5[4:])
+                else:
+                    continue
+                if rmd5 and srcmd5 in rmd5:
+                    skip = True
+                    break
+        if skip:
+            print('\tsame remote file exists ' + os.path.basename(source))
+        else:
+            with open(source) as f:
+                try:
+                    rc = site.action.resource_patch(
+                        id=res['id'],
+                        hash='md5-%s' % base64.b64encode(md5),
```

Comment:
One last change: client requested that users should be able to verify that the files they receive are the right files and md5 is a very weak assurance, especially with binary files like PDFs. While looping over the chunks of the files and calculating the md5, please also calculate the SHA384 and store it in this field like `hash='md5-%s sha384-%s' % (b64encode(md5), b64encode(sha384))`